cognite/neat/rules/exporters/_loaders.py

```diff
@@ -0,0 +1,261 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from graphlib import TopologicalSorter
+from typing import Generic, TypeVar, cast
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes._base import (
+    T_CogniteResourceList,
+    T_WritableCogniteResource,
+    T_WriteClass,
+    WriteableCogniteResourceList,
+)
+from cognite.client.data_classes.data_modeling import (
+    Container,
+    ContainerApply,
+    ContainerApplyList,
+    ContainerList,
+    DataModel,
+    DataModelApply,
+    DataModelApplyList,
+    DataModelingId,
+    DataModelList,
+    RequiresConstraint,
+    Space,
+    SpaceApply,
+    SpaceApplyList,
+    SpaceList,
+    View,
+    ViewApply,
+    ViewApplyList,
+    ViewList,
+)
+from cognite.client.data_classes.data_modeling.ids import (
+    ContainerId,
+    DataModelId,
+    InstanceId,
+    VersionedDataModelingId,
+    ViewId,
+)
+from cognite.client.utils.useful_types import SequenceNotStr
+
+T_ID = TypeVar("T_ID", bound=str | (int | (DataModelingId | (InstanceId | VersionedDataModelingId))))
+T_WritableCogniteResourceList = TypeVar("T_WritableCogniteResourceList", bound=WriteableCogniteResourceList)
+
+
+class ResourceLoader(
```

Comment:
This is steeling a concept from `cdf-toolkit`. However, I thing it is equally useful for us to be able to tell the user results of dry runs and have a standardized interface towards cognite resources.